gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -217,16 +218,44 @@ func (vm *VMKeeper) Call(ctx sdk.Context, msg MsgCall) (res string, err error) {
 	mpn := gno.NewPackageNode("main", "main", nil)
 	mpn.Define("pkg", gno.TypedValue{T: &gno.PackageType{}, V: pv})
 	mpv := mpn.NewPackage()
-	// Parse expression.
+
+	if len(msg.Args) != len(ft.Params) {
+		return "", fmt.Errorf("wrong number of arguments in call to %s: want %d got %d", fnc, len(ft.Params), len(msg.Args))
+	}
+
+	request := make([]gno.TypedValue, len(ft.Params))
+	for i, arg := range msg.Args {
+		pt := ft.Params[i].Type
+		arg = strings.TrimSpace(arg)
+		if len(arg) > 2 &&
```

Comment:
Should this be `>= 2`? Maybe it is sometimes desireable to pass an empty slice or struct.

Reply:
yes, definitely :)